File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/post/PostDTO.java
package pl.dawid0604.pcForum.dto.post;

import pl.dawid0604.pcForum.dto.user.UserProfileDTO;

import java.util.List;

public record PostDTO(String encryptedId, UserProfileDTO user,
                      List<PostContentDTO> content, String createdAt, String lastUpdatedAt,
                      boolean loggedUserPost, long numberOfUpVotes, long numberOfDownVotes,
                      boolean loggedUserHasUpVote, boolean loggedUserHasDownVote) { }

File: frontend/src/app/model/PostDTO.ts
import { PostContentDTO } from "./PostContentDTO"
import { UserProfileDTO } from "./UserProfileDTO"

export interface PostDTO {
    encryptedId: string,
    user: UserProfileDTO,
    createdAt: string,
    content: PostContentDTO[],
    lastUpdatedAt: string,
    loggedUserPost: boolean,
    numberOfUpVotes: number,
    numberOfDownVotes: number,
    loggedUserHasUpVote: boolean,
    loggedUserHasDownVote: boolean
}